pkg/templates/kubeadm/provider.go

```diff
@@ -0,0 +1,66 @@
+/*
+Copyright 2019 The KubeOne Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package kubeadm
+
+import (
+	"github.com/Masterminds/semver"
+	"github.com/pkg/errors"
+
+	kubeoneapi "github.com/kubermatic/kubeone/pkg/apis/kubeone"
+	"github.com/kubermatic/kubeone/pkg/util"
+)
+
+var (
+	v13x = mustParseConstraint("1.13.x")
+	v14x = mustParseConstraint("1.14.x")
+	v15x = mustParseConstraint("1.15.x")
+)
+
+// KubeADM interface abstract differences between different kubeadm versions
+type KubeADM interface {
+	Config(ctx *util.Context, instance kubeoneapi.HostConfig) (string, error)
+	UpgradeLeaderCMD() string
+	UpgradeFollowerCMD() string
+}
+
+// New constructor
+func New(ver string) (KubeADM, error) {
+	sver, err := semver.NewVersion(ver)
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to parse version")
+	}
+	switch {
```

Comment:
I'd like to enforce this check at the validation level too. If you provide not supported version, e.g. `1.16`, it's going to pass validation but fail on install.

Reply:
>but fail on install

Not necessary. As kubeadm interfaces are stabilizing they are becoming more and more forward compatible. In order to avoid failing with 1.15 I'll add v1beta2 kubeadm APIs to this PR.